pkg/secrets/k8s/k8s_secrets.go

```diff
@@ -185,11 +186,12 @@ func generateStringDataEntry(stringDataEntriesMap map[string][]byte) ([]byte, er
 	entries := make([][]byte, len(stringDataEntriesMap))
 	// Parse every key-value pair in the map to a "key:value" byte array
 	for key, value := range stringDataEntriesMap {
+		stringDataEntriesMap[key] = bytes.Replace(value, []byte("\\"), []byte("\\\\"), -1)
```

Comment:
please move this to a separate method and add an explanation there on why we are doing this

Reply:
also, i think that running `bytes.Replace` with -1 is like running `bytes.ReplaceAll`. Am I wrong?

Reply:
i guess this works if the UT is passing but can you explain why we are replacing `\\` with `\\\\` and not `\` with `\\`? seems to me that an input with just 1 backslash will stay that way.

Reply:
why do you update `stringDataEntriesMap[key]` instead of editing `value`? is this map entry used later in the code?

Reply:
A couple of things:
1. It is a one line of code so is it worth throwing it in another method?
2. I didn't have the proper of go version locally (need 1.12 to use replaceAll). Updated the go version on my end and substituted replace -> replaceAll as noted
3. Because go does not escape a single '\', even if in a string and therefore leads to a error "unknown escape character"
4. Didn't want to touch the index value but will change it now

Reply:
1. yes, for the explanation. It improves readability. 
2. thanks. Nice catch with the go version!
3. ok
4. thanks